src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -190,43 +190,94 @@ namespace
           : m_renderer(renderer)
           , m_active_camera(camera)
         {
+            m_callback_key = GetISceneEventManager()->RegisterCallback(this, false, 100, true);
         }
 
-        virtual void ControllerOtherEvent(NodeKeyTab& nodes) override
+        virtual ~SceneChangeCallback()
+        {
+            GetISceneEventManager()->UnRegisterCallback(m_callback_key);
+        }
+
+        virtual void ModelOtherEvent(NodeKeyTab& nodes) override
```

Comment:
This is clearly correct, but our convention is to omit `virtual` when a method is marked as `override` (since it implies `virtual` already).